db/migrate/20210113091207_add_unique_index_on_labels.rb

```diff
@@ -0,0 +1,6 @@
+class AddUniqueIndexOnLabels < ActiveRecord::Migration[5.2]
+  def change
+    remove_index :labels, [:user_id, :key, :scope]
+    add_index :labels, [:user_id, :key, :scope], unique: true
```

Comment:
You will have a problem here if there are already duplicated entries in the database, you can try to write a spec for this migration.